internal/service/service.go

```diff
@@ -0,0 +1,133 @@
+package service
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cast"
+
+	"github.com/carfloresf/the-sun-god/internal/constants"
+	"github.com/carfloresf/the-sun-god/internal/model"
+	"github.com/carfloresf/the-sun-god/pkg/subreddit"
+)
+
+var ErrMutuallyExclusiveFields = fmt.Errorf("link and content are mutually exclusive fields")
```

Comment:
How do you feel about adding custom error type? Then, it can be tested by `errors.As` for example, and you can avoid having global public variable which can cause issues. See https://youtu.be/pN_lm6QqHcw?t=934

Reply:
yeah that would work to, I'm not very used to creating them but I think it is a good suggestion